### PR TITLE
Added possibility to download refs

### PR DESCRIPTION
**Pull request by Lucpen**
## PR checklist

- [x] Added possibility to download references.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/genomic-medicine-sweden/tomte/tree/master/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,29 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+# X.X.X - [XXXX-XX-XX]
+
+### `Added`
+
+- Fasta, gtf, vep cache and plugins can now be downloaded automatically by the pipeline if they are not provided by the user [#144](https://github.com/genomic-medicine-sweden/tomte/pull/144)
+- Added `--gencode_annotation_version`, the version of the gencode reference version to download if fasta or gtf is not provided
+- Added the possibility to provide `--vep_refs_download`, a comma separated csv determining the vep references that should be downloaded (excluding gnomad ones) alongside with a switch `--skip_download_vep` for the vep reference download in general and `--skip_download_gnomad` for gnomad in particular[#149](https://github.com/genomic-medicine-sweden/tomte/pull/144)
```
Comment (Lucpen) [inline CHANGELOG.md:12]:
```suggestion
- Added `--gencode_annotation_version`, the version of the gencode reference version to download if fasta or gtf is not provided [#149](https://github.com/genomic-medicine-sweden/tomte/pull/149)
- Added the possibility to provide `--vep_refs_download`, a comma separated csv determining the vep references that should be downloaded (excluding gnomad ones) alongside with a switch `--skip_download_vep` for the vep reference download in general and `--skip_download_gnomad` for gnomad in particular [#149](https://github.com/genomic-medicine-sweden/tomte/pull/149)
```

Comment (Lucpen) [inline CHANGELOG.md:12]:
```suggestion
- Fasta, gtf, vep cache and plugins can now be downloaded automatically by the pipeline if they are not provided by the user [#149](https://github.com/genomic-medicine-sweden/tomte/pull/149)
- Added `--gencode_annotation_version`, the version of the gencode reference version to download if fasta or gtf is not provided [#149](https://github.com/genomic-medicine-sweden/tomte/pull/149)
- Added the possibility to provide `--vep_refs_download`, a comma separated csv determining the vep references that should be downloaded (excluding gnomad ones) alongside with a switch `--skip_download_vep` for the vep reference download in general and `--skip_download_gnomad` for gnomad in particular [#149](https://github.com/genomic-medicine-sweden/tomte/pull/149)
```

**Review by jemten — COMMENTED** [review]
Massive work Lucia. A couple of comments :)

File: conf/modules/prepare_references.config [hunk 2]
```diff
@@ -17,7 +17,7 @@
 process {
 
     withName: '.*PREPARE_REFERENCES:GUNZIP_FASTA' {
-        ext.when = {params.fasta.endsWith(".gz")}
+        ext.when = { params.fasta && params.fasta?.endsWith(".gz") }
```
[the remark this answers is not in the document]
Comment (Lucpen) [reply conf/modules/prepare_references.config:20]: If params.fasta is there, does it ends with .gz?

[the remark this answers is not in the document]
Comment (jemten) [reply conf/modules/prepare_references.config:20]: But do you need `?` ? Looks like you're doing fine without it on line 137 in this file

[the remark this answers is not in the document]
Comment (Lucpen) [reply conf/modules/prepare_references.config:20]: you are right, it is not necessary

Comment (Lucpen) [inline conf/modules/prepare_references.config:20]:
```suggestion
        ext.when = { params.fasta && params.fasta.endsWith(".gz") }
```

File: conf/modules/prepare_references.config [hunk 3]
```diff
@@ -51,7 +51,7 @@ process {
     }
 
     withName: '.*PREPARE_REFERENCES:GUNZIP_GTF' {
-        ext.when = { params.gtf.endsWith(".gz") }
+        ext.when = { params.gtf && params.gtf?.endsWith(".gz") }
```
Comment (Lucpen) [inline conf/modules/prepare_references.config:54]:
```suggestion
        ext.when = { params.gtf && params.gtf.endsWith(".gz") }
```

File: test_data/vep_to_download.csv [hunk 4]
```diff
@@ -0,0 +1,4 @@
+name,path_for_wget
+clinvar_20240806.vcf.gz,ftp://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh37/weekly/clinvar.vcf.gz
+clinvar_20240806.vcf.gz.tbi,ftp://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh37/weekly/clinvar.vcf.gz.tbi
+homo_sapiens_merged,https://ftp.ensembl.org/pub/release-110/variation/indexed_vep_cache/gorilla_gorilla_refseq_vep_110_gorGor4.tar.gz
```
Comment (Lucpen) [inline test_data/vep_to_download.csv:4]:
```suggestion
homo_sapiens_merged,https://ftp.ensembl.org/pub/release-110/variation/indexed_vep_cache/gorilla_gorilla_refseq_vep_110_gorGor4.tar.gz

```

File: workflows/tomte.nf [hunk 5]
```diff
@@ -79,24 +98,25 @@ workflow TOMTE {
...
+    }
+    .map { f ->
+        if (params.skip_download_vep) {
+            if(f.isFile() || f.isDirectory()) {
+                return f
             } else {
                 error("\nVep database file ${f} does not exist.")
             }
+        } else {
+            return f
```
Comment (Lucpen) [inline workflows/tomte.nf:115]:
```suggestion
    .map { path ->
        if (params.skip_download_vep) {
            if(path.isFile() || path.isDirectory()) {
                return path
            } else {
                error("\nVep database file ${path} does not exist.")
            }
        } else {
            return path
```

File: test_data/vep_to_download.csv (hunk 4 above)
Comment (Lucpen) [inline test_data/vep_to_download.csv:4]:
```suggestion
homo_sapiens_merged,https://ftp.ensembl.org/pub/release-110/variation/indexed_vep_cache/homo_sapiens_merged_vep_110_GRCh37.tar.gz

```

File: .nf-core.yml [hunk 6]
```diff
@@ -21,6 +21,7 @@ lint:
     - lib/NfcoreTemplate.groovy
     - docs/README.md
     - pyproject.toml
+    - .prettierignore
```
Comment (Lucpen) [inline .nf-core.yml:24]:
```suggestion
```

File: .prettierignore [hunk 7]
```diff
@@ -3,7 +3,7 @@ adaptivecard.json
 slackreport.json
 .nextflow*
 work/
-data/
+test_data/*
```
Comment (Lucpen) [inline .prettierignore:6]:
```suggestion
data/
```

**Review by jemten — APPROVED** [review]
🚀

**Pull request merged**